AnsibleTestCRD.yaml

```
@@ -0,0 +1,33 @@
+---
+apiVersion: test.openstack.org/v1beta1
+kind: AnsibleTest
+metadata:
+  name: rhel-ai-validation-ansibletest
+  namespace: openstack
+spec:
+  ansibleGitRepo: https://github.com/openstack-k8s-operators/rhelai-validation.git
+  # yamllint disable rule:line-length
+  ansibleInventory: |
+    [control_node]
+    localhost ansible_connection=local
+    [target_nodes]
+    rhelai0 ansible_host=192.168.122.222 ansible_user=cloud-user ansible_ssh_private_key_file=~/rhel-ai.pem ansible_ssh_common_args='-o StrictHostKeyChecking=no'
+  # yamllint enable rule:line-length
+  ansiblePlaybookPath: main.yaml
+  ansibleVarFiles: |
+    deploy_rhelai_private_key_file: ~/rhel-ai.pem
```

The key specified in `computeSSHKeySecretName` is mounted to `/var/lib/ansible/.ssh/compute_id`.

```suggestion
    deploy_rhelai_private_key_file: /var/lib/ansible/.ssh/compute_id
```

---

Ah, great, thank you! I will experiment with this and add the required `oc create secret` command out of the CRD comments and into the README.

---

I got a bit carried away and wrote some improved SSH key handling here (totally untested): https://github.com/csibbitt/rhelai-validation/commit/88e5f3f6ec37d75966d51bfc98b0517399c01f53 

It turned out to be a bit involved (especially to have it work both on the hypervisor and in openstackclient pod) and it's going to take me some more hours to test it properly, so I'd like to handle it as a separate PR and get started on some of the JunitXML stuff first.

If I can't get to it all this week, then I'll open a ticket to finish it up next sprint since it fits more into test-operator compatibility than JunitXML / refactoring into a role.

---

IMO, we can replace the `~/rhel-ai.pem` with `/var/lib/ansible/.ssh/compute_id` (change of two lines) and then address this in a separate PR when this becomes ready -> https://github.com/csibbitt/rhelai-validation/commit/88e5f3f6ec37d75966d51bfc98b0517399c01f53

If we merge it as it is right now, then the Ansible CR does not work.  I'm marking this Unresolved. But if you want to merge this, I'm fine with it.

---

What do you mean by "Ansible CR does not work", can you be more specific? It works for me, see https://github.com/openstack-k8s-operators/rhelai-validation/pull/3#issuecomment-2695601585

In fact, I made the change you suggest[1], and now it doesn't work:

```
$ oc logs -f rhel-ai-validation-ansibletest-gg28p

[...]
TASK [rhelai-validation : Write ssh key to file for rhel-ai] *******************
task path: /var/lib/ansible/ansible/rhelai-validation/tasks/vm_deploy.yaml:10
[...]

fatal: [localhost]: FAILED! => {
[...]
    "msg": "Destination /var/lib/ansible/.ssh not writable"
}

PLAY RECAP *********************************************************************
localhost                  : ok=4    changed=4    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0 
```


[1] https://github.com/csibbitt/rhelai-validation/commit/c5d32c157351181104c47a2d4549baadeffcac67

I experimented with this yesterday, found a possible solution, and will need to find time to test it in a future sprint. I've written OSPRH-14461 to track this in case I can't get to it this week.

---

My apologies. I've completely forgotten this morning that we are creating the private key in rhelai-validation [1]. I lived in the universe where rhoso-rhelai is still responsible for deployment of the VM we consume and the private key as well. 

You are right, it works. I'm marking this as resolved.

[1] https://github.com/openstack-k8s-operators/rhelai-validation/blob/a6b1f827a9a87c46f4ebb8f43b4796e4e1dd6570/rhelai-validation/tasks/deploy_vm.yaml#L10